Lib/test/test_os.py

```diff
@@ -126,11 +126,21 @@ def test_closerange(self):
     @support.cpython_only
     def test_rename(self):
         path = support.TESTFN
+        not_exists = 'temp2/test'
         old = sys.getrefcount(path)
         self.assertRaises(TypeError, os.rename, path, 0)
         new = sys.getrefcount(path)
         self.assertEqual(old, new)
 
+    def test_rename_bad_dir(self):
+        path1 = 'temp/not-exists'
+        path1new = 'temp/test2/test3/test4'
+        try:
+            os.rename(path1, path1new)
```

Comment:
I don't understand your test. Do you **expect** FileNotFoundError? If yes, you have to use:

```
with self.assertRaises(FileNotFoundError):
    os.rename(path1, path1new)
self.assertFalse(path.exists(path1new))
```

Currently, "self.assertFalse(path.exists(path1new))" is never executed if I understand correctly!

Reply:
Actually you are testing `os.rename`, not `os.renames`. Also (unrelated to this change per se):
1) it appears there are no tests for `os.renames` at all
2) also `os.rename` test isn't actually testing the actual functionality

I will file a ticket for that but I would recommend  having them before proceeding with this.